.github/workflows/go-pull-request.yml

```
@@ -0,0 +1,242 @@
+name: C++ Code Analysis
+# this runs the workflow on PR and make sure that the code is tested and analyzed
+
+on:
+  push:
+    branches: ["develop"]
+    paths:
+      - 'sdks/go/**'
+      #- 'unittests/go/**'
+      #- 'tools/codegen/**'
+  pull_request:
+    paths:
+      - 'sdks/go/**'
+      #- 'unittests/go/**'
+      #- 'tools/codegen/**'
+      - '.github/workflows/go-pull-request.yml'
+  workflow_dispatch:
+
+jobs:
+  Trivy:
+    name: Trivy vulnerability scanner (25 seconds)
+    timeout-minutes: 2
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+
+      - name: Run Trivy vulnerability scanner in repo mode
+        uses: aquasecurity/trivy-action@0.28.0
+        with:
+          scan-type: 'fs'
+          ignore-unfixed: true
+          format: 'sarif'
+          output: 'trivy-results.sarif'
+          severity: 'HIGH,CRITICAL'
+          skip-dirs: '.git,build,coverage,docs,sdks/java,sdks/cpp/unittests'
```

```suggestion
          skip-dirs: '.git,build,coverage,docs,sdks/java,unittests'
```
also need to go and add the go sdk to the `skip-dirs` on the cpp workflow

---

will add to cpp. But we definitely want to include sdks/cpp to the skiplist here no?

---

OH! absolutely lol, whoops!